### PR TITLE
LXD topology support

### PR DESCRIPTION
**Pull request by panfantastic**
This is a first attempt at introducing LXD topology support.

I welcome any and all criticisms... Including my rather copy-paste affair from the docker plugin.  Please at least be constructive ;)

What do I need to do extra to get this feature merged?

### CODE REVIEW

**Comment by lebauce** [comment]
ok to test

**Comment by lebauce** [comment]
@panfantastic Thanks for your PR ! We welcome any contribution and we try to be as constructive as possible :-)

First, I'm not familiar with LXD so we'll have to give it a try.

It seems your PR relies on https://github.com/lxc/lxd, so you'll need to add it to the vendor/vendor.json files (with `govendor fetch github.com/lxc/lxd`)

It would be really nice to have it tested on our CI. We can definitely help you on this. You could add a `lxd_test.go` in the `tests` folder, mimicking the Docker tests in `tests/topology_test.go`. In the meantime, I can set up lxd on our infrastructure.

We can also discuss about it on IRC (#skydive-project on FreeNode) or on https://chat.skydive.network

**Comment by panfantastic** [comment]
Hi @lebauce 

Thank you for your very welcoming response :D 

I'm looking at adding tests soon, now that I know where they were hiding ;)  

As to LXD version - I've been using version 2.17 in production at work and at home.  The PR however is updated to be vendored against the tag for 2.21.  

Spinning up LXD containers can be a bit finickity for defining some config on the cli (at work we use custom Ansible roles to do all the heavy lifting), so I'm not sure how simple it will be to write the tests or where I can simplify the requirements for testing.  I'll see what I can do though.


Many Thanks
Iain

**Comment by lebauce** [comment]
@panfantastic Our CI machines are running Fedora 25 but the COPR repository for LXD is only for Fedora 26 and 27. So I upgraded them. I'll then set up LXD on them.

Could you please your PR on master ? (we recently changed the way we use etcd on our CI and your PR still uses the old way).

**Comment by lebauce** [comment]
@panfantastic In general, we try to avoid polling when possible. I gave the LXD events API a try. While it's a bit tricker than for Docker, it works and we are notified of container creation/deletion.

I pushed my attempt there : https://github.com/lebauce/skydive/tree/lxd-events (the 2 last commits).
Feel free to squash this into your PR.

We should also modify the default configuration file (`etc/skydive.yml.default`) and add a picture to the Web UI (create a new `lxd.png` `statics/img/` and modify the `managerImgMap` in `statics/js/components/layout.js`).

With this, we'll be very close to have this merged :-)

**Comment by panfantastic** [comment]
Thanks, I've not had time over the weekend for this.  I am happy to give you write access to my branch, rather than squashing your commits in, given you're going out of your way.

File: topology/probes/lxd/lxd.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
+// +build linux
+
+package lxd
+
+import (
+	"fmt"
+	"github.com/vishvananda/netns"
+//	"github.com/kr/pretty"
```
Comment (nplanel) [inline topology/probes/lxd/lxd.go]: please remove commented import

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-orientdb

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-orientdb

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run skydive-scale-tests

**Comment by lebauce** [comment]
@panfantastic Could you please this version a try and tell me if it still works for you. Thanks

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-orientdb

**Pull request merged**